x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation_logic.test.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  LogicMounter,
+  mockHttpValues,
+  mockKibanaValues,
+  mockFlashMessageHelpers,
+} from '../../../__mocks__';
+
+import { nextTick } from '@kbn/test/jest';
+import { generatePath } from 'react-router-dom';
+
+import { ENGINE_PATH } from '../../routes';
+import { formatApiName } from '../../utils/format_api_name';
+
+import { ENGINE_CREATION_SUCCESS_MESSAGE } from './constants';
+import { EngineCreationLogic } from './engine_creation_logic';
+
+describe('EngineCreationLogic', () => {
+  const { mount } = new LogicMounter(EngineCreationLogic);
+  const { http } = mockHttpValues;
+  const { navigateToUrl } = mockKibanaValues;
+  const { setQueuedSuccessMessage, flashAPIErrors } = mockFlashMessageHelpers;
+
+  const DEFAULT_VALUES = {
+    name: '',
+    rawName: '',
+    language: 'Universal',
+  };
+
+  it('has expected default values', () => {
+    mount();
+    expect(EngineCreationLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('setLanguage', () => {
+      it('sets language to the provided value', () => {
+        mount();
+        EngineCreationLogic.actions.setLanguage('English');
+        expect(EngineCreationLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          language: 'English',
+        });
+      });
+    });
+
+    describe('setRawName', () => {
+      beforeAll(() => {
+        mount();
+        EngineCreationLogic.actions.setRawName('Name__With#$&*%Special--Characters');
+      });
+
+      describe('rawName', () => {
+        it('should be set to provided value', () => {
+          expect(EngineCreationLogic.values.rawName).toEqual('Name__With#$&*%Special--Characters');
+        });
+      });
+
+      describe('name', () => {
+        it('should be set to a sanitized value', () => {
+          expect(EngineCreationLogic.values.name).toEqual('name-with-special-characters');
+        });
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('onEngineCreationSuccess', () => {
+      beforeAll(() => {
+        mount({ language: 'English', rawName: 'test' });
+        EngineCreationLogic.actions.onEngineCreationSuccess();
+      });
+
+      it('should set a success message', () => {
+        expect(setQueuedSuccessMessage).toHaveBeenCalledWith('Successfully created engine.');
```

Comment:
I don't see `jest.clearAllMocks` being called anywhere in this file.

If they're not reset, I worry that these tests unintentionally becoming order dependent.

Reply:
Like when we do stuff like this:
```
      beforeAll(() => {
        mount({ language: 'English', rawName: 'test' });
      });
```

I feel like maybe we need to add the following as well
```
      afterAll(() => {
        jest.ClearAllMocks();
      });
```

I know it's not as efficient but lately I've been thinking the cleanest way to write tests is to avoid `beforeAll` and clear everything and start fresh with each test. It's less efficient when running, but can save development time and headaches.

That last thought is just me thinking out loud, you don't need to change it.